source/developers/in-context-editing-ftl.rst

```diff
@@ -0,0 +1,368 @@
+
+The previous template markup for Studio support still works however the new markup generates cleaner code and exposes new features.
+
+.. note::
+   If your ``cstudio-support.ftl`` uses ``siteContext.overlayCallback`` to check if Engine is running in preview mode, please update your file to use ``modePreview`` instead as ``siteContext.overlayCallback`` and related classes are being discontinued in Engine.
+
+     Search for ``siteContext.overlayCallback`` calls in your ``cstudio-support.ftl`` file:
+
+     .. code-block:: guess
+        :caption: cstudio-support.ftl
+        :emphasize-lines: 2
+
+        <#macro toolSupport>
+          <#if siteContext.overlayCallback??>
+            <script src="/studio/static-assets/libs/requirejs/require.js" data-main="/studio/overlayhook?site=NOTUSED&page=NOTUSED&cs.js"></script>
+            <script>document.domain = "${Request.serverName}"; </script>
+          </#if>
+
+     |
+
+     Replace with ``modePreview`` to check if Engine is running in preview mode:
+
+     .. code-block:: guess
+        :caption: cstudio-support.ftl
+        :emphasize-lines: 2
+
+        <#macro toolSupport>
+          <#if modePreview>
+            <script src="/studio/static-assets/libs/requirejs/require.js" data-main="/studio/overlayhook?site=NOTUSED&page=NOTUSED&cs.js"></script>
+            <script>document.domain = "${Request.serverName}"; </script>
+          </#if>
+
+     |
+
+--------------------------
+Enabling Authoring Support
+--------------------------
+
+At the top of your page or component (whatever it is you are rendering, include the following) import:
+
+    .. code-block:: guess
+
+	    <#import "/templates/system/common/cstudio-support.ftl" as studio/>
+
+|
+
+At the bottom of your template insert the following: (Note the example shows a traditional HTML page however other formats/levels of granularity are supported
+
+    .. code-block:: guess
+
+	        <@studio.toolSupport/>
+	      </body>
+	    </html>
+
+|
+
+--------------------------
+In-Context Editing Pencils
+--------------------------
+
+In context editing renders pencils on the screen that invoke editing controls when clicked.  This allows authors to quickly/visually identify editable content and make changes.
+
+.. image:: /_static/images/ice-example.png
+        :align: center
+        :width: 70 %
+        :alt: In context editing example
+
+|
+
+To enable in-context editing simply add the following attribute to the container/element where you want to place the editing control
+
+    .. code-block:: guess
+
+	    <@studio.iceAttr iceGroup="author"/>
+
+|
+
+For embedded components, to enable in-context editing simply add the following attribute and tag attribute to the embedded component template
```

Comment:
the `component` parameter is not only for embedded anymore, shared components should use it too